src/GuiBase/BaseApplication.cpp

```diff
@@ -356,77 +362,75 @@ void BaseApplication::addBasicShaders() {
 
-    // ----------
-    // 2. Kickoff rendering
-    m_viewer->startRendering( dt );
+        // ----------
+        // 2. Kickoff rendering
+        m_viewer->startRendering( dt );
 
-    timerData.tasksStart = Core::Utils::Clock::now();
+        timerData.tasksStart = Core::Utils::Clock::now();
 
-    // ----------
-    // 3. Run the engine task queue.
-    m_engine->getTasks( m_taskQueue.get(), dt );
+        // ----------
+        // 3. Run the engine task queue.
+        m_engine->getTasks( m_taskQueue.get(), dt );
 
-    if ( m_recordGraph ) { m_taskQueue->printTaskGraph( std::cout ); }
+        if ( m_recordGraph ) { m_taskQueue->printTaskGraph( std::cout ); }
 
-    // Run one frame of tasks
-    m_taskQueue->startTasks();
-    m_taskQueue->waitForTasks();
-    timerData.taskData = m_taskQueue->getTimerData();
-    m_taskQueue->flushTaskQueue();
+        // Run one frame of tasks
+        m_taskQueue->startTasks();
+        m_taskQueue->waitForTasks();
+        timerData.taskData = m_taskQueue->getTimerData();
+        m_taskQueue->flushTaskQueue();
 
-    timerData.tasksEnd = Core::Utils::Clock::now();
+        timerData.tasksEnd = Core::Utils::Clock::now();
 
-    // ----------
-    // 4. Wait until frame is fully rendered and display.
-    m_viewer->waitForRendering();
+        // ----------
+        // 4. Wait until frame is fully rendered and display.
+        m_viewer->waitForRendering();
 
-    timerData.renderData = m_viewer->getRenderer()->getTimerData();
+        timerData.renderData = m_viewer->getRenderer()->getTimerData();
 
-    // ----------
-    // 5. Synchronize whatever needs synchronisation
-    m_engine->endFrameSync();
+        // ----------
+        // 5. Synchronize whatever needs synchronisation
+        m_engine->endFrameSync();
 
-    // ----------
-    // 6. Frame end.
-    timerData.frameEnd = Core::Utils::Clock::now();
-    timerData.numFrame = m_frameCounter;
+        // ----------
+        // 6. Frame end.
+        timerData.frameEnd = Core::Utils::Clock::now();
+        timerData.numFrame = m_frameCounter;
 
-    if ( m_recordTimings ) { timerData.print( std::cout ); }
+        if ( m_recordTimings ) { timerData.print( std::cout ); }
 
-    m_timerData.push_back( timerData );
+        m_timerData.push_back( timerData );
 
-    if ( m_recordFrames ) { recordFrame(); }
+        if ( m_recordFrames ) { recordFrame(); }
 
-    ++m_frameCounter;
+        ++m_frameCounter;
 
-    if ( m_numFrames > 0 && m_frameCounter > m_numFrames ) { appNeedsToQuit(); }
+        if ( m_numFrames > 0 && m_frameCounter > m_numFrames ) { appNeedsToQuit(); }
 
-    if ( m_frameCounter % m_frameCountBeforeUpdate == 0 )
-    {
-        emit( updateFrameStats( m_timerData ) );
-        m_timerData.clear();
+        if ( m_frameCounter % m_frameCountBeforeUpdate == 0 )
+        {
+            emit( updateFrameStats( m_timerData ) );
+            m_timerData.clear();
+        }
     }
-
     m_mainWindow->onFrameComplete();
```

Comment:
Are you sure this must be called outside of the frame update block (the if above) ?

Reply:
Oops my comments were not taken into account when I click resolve conversation. 
So here again : well for this one I don't know but I remark I double check to update, so I simply remove the test here.